src/lib.rs

```diff
@@ -169,12 +199,20 @@ impl Monado {
 
 		Ok(Device {
 			monado: self,
-			index: index as u32,
+			index,
 			id,
 			name,
 		})
 	}
 
+	pub fn device_index_from_role(&self, role: DeviceRole) -> Result<u32, MndResult> {
```

Comment:
exporting this is nice because you want to enumerate all devices anyway, even ones that don't have a role, and this way you can just assign roles based on only the indices you get from this.